src/pudl/extract/vceregen.py

```diff
@@ -0,0 +1,183 @@
+        """Initialize the module.
+
+        Args:
+            ds (:class:datastore.Datastore): Initialized datastore.
+        """
+        super().__init__(*args, **kwargs)
+        self._file_name = self._load_csv(self._pkg, "file_map.csv")
+
+    def get_all_pages(self) -> list[str]:
+        """Hard code the page names, which usually are pulled from column rename spreadsheets."""
+        return VCEREGEN_PAGES
+
+    def get_file_name(self, page, **partition):
+        """Returns file name of given partition and page."""
+        return self._file_name.loc[page, str(self._get_partition_selection(partition))]
+
+
+class Extractor(CsvExtractor):
+    """Extractor for VCE renewable generation profiles."""
+
+    def __init__(self, *args, **kwargs):
+        """Initialize the module.
+
+        Args:
+            ds (:class:datastore.Datastore): Initialized datastore.
+        """
+        self.METADATA = VCEMetadata("vceregen")
+        super().__init__(*args, **kwargs)
+
+    def get_column_map(self, page, **partition):
+        """Return empty dictionary, we don't rename these files."""
+        return {}
+
+    def source_filename(self, page: str, **partition: PartitionSelection) -> str:
+        """Produce the CSV file name as it will appear in the archive.
+
+        The files are nested in an additional folder with the year name inside of the
+        zipfile, so we add a prefix folder based on the yearly partition to the source
+        filename.
+
+        Args:
+            page: pudl name for the dataset contents, eg "boiler_generator_assn" or
+                "coal_stocks"
+            partition: partition to load. Examples:
+                {'year': 2009}
+                {'year_month': '2020-08'}
+
+        Returns:
+            string name of the CSV file
+        """
+        return f"{partition['year']}/{self._metadata.get_file_name(page, **partition)}"
+
+    def load_source(self, page: str, **partition: PartitionSelection) -> pd.DataFrame:
+        """Produce the dataframe object for the given partition.
+
+        Args:
+            page: pudl name for the dataset contents, eg "boiler_generator_assn" or
+                "data"
+            partition: partition to load. Examples:
+                {'year': 2009}
+                {'year_month': '2020-08'}
+
+        Returns:
+            pd.DataFrame instance containing CSV data
+        """
+        with (
+            self.ds.get_zipfile_resource(self._dataset_name, **partition) as zf,
+        ):
+            # # Get path to zipfile
+            # zippath = zf.filename
+            # Get list of file names in the zipfile
+            files = zf.namelist()
+            # Get the particular file of interest
+            file = next(
+                (x for x in files if self.source_filename(page, **partition) in x), None
+            )
+            # # Read it in using dask
+            df = pd.read_csv(BytesIO(zf.read(file)), **self.READ_CSV_KWARGS)
```

Comment:
I think there are some tricks you can play with the keyword args to `read_csv()` if you need to reduce memory usage. Like if you can give it a specific and smaller dtype (float32) to cast all the data to on read, rather than doing it after the fact, using `low_memory=True` and giving it a chunksize so that the textual data is read and re-cast incrementally, you might try `engine="pyarrow"` and `dtype_backend="pyarrow"`

Reply:
@zaneselvans Definitely, I was just trying to get a functional version out asap and I still have a bit of clean up to do. It seems like you're hesitant about using dask instead of pandas - is there a reason why this (and eventually cems for similar reasons) wouldn't be better use cases for dask than pandas?

Reply:
Well, it's another layer of complexity and it has a lot of overlap with the problems that Dagster is trying to solve -- building a DAG that defines dependencies, and respecting them when doing the computation in parallel.  AFAIK there isn't a good way within Dagster to manage this kind of nested parallelism, and it's caused us trouble in some assets where we've got multi-threaded linear algebra happening through Numpy.

We could use Dagster's Dask executor rather than doing "local" execution both locally and in Google Batch, and maybe that would gracefully handle both kinds of parallelism, since the Dask cluster would be handling both? But I feel like we tested non-local execution and found that with our large number of smaller assets the overhead of distributed processing actually meant the whole ETL ran significantly slower.

Or are you thinking about using Dask to manage serialization of the CSV processing? That might work well for a fast but memory-intensive process where we don't really need to parallelize it to run faster, but we do need to reduce peak resource utilization.

Reply:
I was thinking about this mostly through the extraction and transform steps, since we're effectively doing a small number of quick transforms (pivot, convert datetime) on a giant dataset where the parts don't depend much on one another (e.g., hours 10-20 and hours 21-30 of the year will want the same kind of transforms, and dtypes are consistent throughout the dataset). So yes, more to reduce peak memory than to improve speed.

It looks as simple as setting up a client config and passing a [dask_resource](https://docs.dagster.io/deployment/guides/dask) to dagster, but always simpler said than done ofc. I'll take a closer look at peak memory when I finish the archiver and see what things look like at peak.

Reply:
Decided this is out of scope for this PR.